lib/Sema/CSGen.cpp

```diff
@@ -1937,40 +1937,23 @@ namespace {
 
     Type visitUnresolvedSpecializeExpr(UnresolvedSpecializeExpr *expr) {
       auto baseTy = CS.getType(expr->getSubExpr());
+      Type specializeTy = baseTy;
 
-      if (baseTy->isTypeVariableOrMember()) {
-        return baseTy;
+      if (AnyMetatypeType *meta = baseTy->getAs<AnyMetatypeType>()) {
+        specializeTy = meta->getInstanceType();
```

Comment:
We have a way to look through metatypes on `TypeBase` - `getMetatypeInstanceType()`.

Reply:
Fixed!